src/controllers/ScheduleGenerator.js

```diff
@@ -0,0 +1,452 @@
+                
+                if (match === 0) {
+                    roundMatches.push([n - 1, away]);
+                } else {
+                    roundMatches.push([home, away]);
+                }
+            }
+            schedule.push(roundMatches);
+        }
+    
+        const groupTeams = {};
+        for (const [groupName, _] of groups) {
+            groupTeams[groupName] = await Team.find({ group: groupName }).exec();
+        }
+    
+        for (let round = 0; round < schedule.length; round++) {
+            for (const [home, away] of schedule[round]) {
+                for (const [groupName, _] of groups) {
+                    await this.createGame({
+                        opponents: [
+                            groupTeams[groupName][home]._id,
+                            groupTeams[groupName][away]._id
+                        ],
+                        gamePhase: 'Group_Stage',
+                        displayName: this.config.display.game_phase_names.group_stage,
+                        duration: this.settings.gameDurationGroupStage / (1000 * 60),
+                        round: round + 1
+                    });
+                }
+            }
+        }
+    
+        return this.getCurrentEndTime();
+    }
+    async generateQuarterFinals() {
+        const matchups = this.config.knockout_stage.quarterfinals.matchups;
+        let quarterFinalNumber = 1;
+
+        for (const [[group1, pos1], [group2, pos2]] of matchups) {
+            const team1 = await this.createDummyTeam(`${pos1}. aus Gruppe ${group1}`, group1);
+            const team2 = await this.createDummyTeam(`${pos2}. aus Gruppe ${group2}`, group2);
+
+            await this.createGame({
+                opponents: [team1, team2],
+                gamePhase: `Quarterfinals ${quarterFinalNumber}`,
+                displayName: `${this.config.display.game_phase_names.quarterfinals} ${quarterFinalNumber}`,
+                duration: this.settings.gameDurationQuarterfinals / (1000 * 60)
+            });
+            quarterFinalNumber++;
+        }
+
+        return this.getCurrentEndTime();
+    }
+
+    async generateSemiFinals() {
+        const matchups = this.config.knockout_stage.semifinals.matchups;
+        let semiFinalNumber = 1;
+
+        for (const [[phase1, num1], [phase2, num2]] of matchups) {
+            const team1 = await this.createDummyTeam(`Sieger ${phase1} ${num1}`, "-");
+            const team2 = await this.createDummyTeam(`Sieger ${phase2} ${num2}`, "-");
+
+            await this.createGame({
+                opponents: [team1, team2],
+                gamePhase: `Semifinals ${semiFinalNumber}`,
+                displayName: `${this.config.display.game_phase_names.semifinals} ${semiFinalNumber}`,
+                duration: this.settings.gameDurationSemifinals / (1000 * 60)
+            });
+            semiFinalNumber++;
+        }
+
+        return this.getCurrentEndTime();
+    }
+
+    async generateFinals() {
```

Comment:
Nice !
I think we had the Finals 1 (WinnerFinale) at the End of the Schedule and the Finals Order: Finals 7, Finals 6, Semifinals 1, Semifinals 2, Finals 4, Finals 3, Finals 2, Finals 8, Finals 1 so that the LastPlaceGame is at the end
You can look at the current server version in the Schedule, but we can finetune the order together with the Soccerexperts